cpp/src/io/comp/gpuinflate.cu

```diff
@@ -1220,5 +1224,41 @@ void gpu_copy_uncompressed_blocks(device_span<device_span<uint8_t const> const>
   }
 }
 
+void decompress_check(device_span<decompress_status> stats,
+                      bool* d_any_block_failure,
+                      rmm::cuda_stream_view stream)
+{
+  if (stats.empty()) { return; }  // Early exit for empty stats
+
+  auto block_failures = rmm::device_uvector<bool>(stats.size(), stream);
+  thrust::transform(rmm::exec_policy(stream),
```

Comment:
There it is! I was looking for a transform_reduce and completely forgot we got iterators for this. Thanks!

Reply:
The second reduce fails with error 98 (`cudaErrorInvalidDeviceFunction`) when I replace the input iterator with
```
  auto failed_status_it = thrust::make_transform_iterator(
    stats.data(), [] __device__(auto stat) -> bool { return stat.status != 0; });
```
Is there a known cub limitation?

Reply:
Changing the return type of this lambda to `int` fixes the issue, but it does not seem correct.

Reply:
Does the problem still occur if you write a functor rather than a lambda?

Reply:
@bdice is correct. The problem is the device lambda. Use a function object.

Reply:
I stumbled on that functor vs lambda issue many times. @jrhemstad Should that be the expected behavior? Or just a compiler limitation?

Reply:
It is an expected limitation of `__device__` lambdas.

Reply:
Should we push the compiler team for removing that limitation? A functor seems to do the same thing as a lambda, right? But if using a lambda, we can embed it right into the code instead of having to define a functor somewhere else far away, which is difficult to read code.

Reply:
Suffice to say that it is _complicated_ and it's not going to change anytime soon.